test/unit/image-inspector.spec.ts

```diff
@@ -0,0 +1,156 @@
+      const result = getImageNameWithIndexDigest(indexDigest, targetImage);
+      expect(result).toEqual(expectedResult);
+    },
+  );
+});
+
+const manifestDigestCases = [
+  [undefined, "nginx:latest", undefined],
+  [
+    "sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+    "nginx@sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+    "nginx@sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+  ],
+  [
+    "sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+    "gcr.io/nginx@sha256:b2aa76581b4388a2138869f7e7d777eaa1c8cc13352c712593466711d30f1b95",
+    "gcr.io/nginx@sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+  ],
+  [
+    "sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+    "nginx:latest",
+    "nginx@sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+  ],
+  [
+    "sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+    "gcr.io/nginx:1.23.2",
+    "gcr.io/nginx@sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+  ],
+  [
+    "sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+    "nginx:1.23.2",
+    "nginx@sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+  ],
+  [
+    "sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+    "nginx@sha256:1a9df42d52d79f0fb19170ee73c61979bec11e1e33b730beaf2d0c6ec333a0f9",
+    "nginx@sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+  ],
+];
+
+describe("getImageNameWithManifestDigest", () => {
+  test.each(manifestDigestCases)(
+    "given manifestDigest %p, target image %p, expect result to be %p",
+    (manifestDigest, targetImage, expectedResult) => {
+      const result = getImageNameWithManifestDigest(
+        manifestDigest,
+        targetImage,
+      );
+      expect(result).toEqual(expectedResult);
+    },
+  );
+});
+
+const imageNamesCases = [
+  [undefined, undefined, "nginx:latest", ["nginx:latest"]],
+  [
+    "nginx@sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+    "nginx@sha256:b2aa76581b4388a2138869f7e7d777eaa1c8cc13352c712593466711d30f1b95",
+    "nginx@sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+    [
+      "nginx@sha256:9604d5d228cf1ba638a767b0d879b600cf288c5aecd68c8b35e30911aadf0dab",
+      "nginx@sha256:b2aa76581b4388a2138869f7e7d777eaa1c8cc13352c712593466711d30f1b95",
+    ],
+  ],
+  [
+    undefined,
+    "nginx@sha256:b2aa76581b4388a2138869f7e7d777eaa1c8cc13352c712593466711d30f1b95",
+    "nginx:latest",
+    [
+      "nginx@sha256:b2aa76581b4388a2138869f7e7d777eaa1c8cc13352c712593466711d30f1b95",
+      "nginx:latest",
+    ],
+  ],
+];
+
+describe("getImageNames", () => {
+  test.each(imageNamesCases)(
+    "given manifestDigest %p, target image %p, expect result to be %p",
```

Comment:
Nit: this prints the wrong data (3 substitutions, but 4 arguments passed)

Reply:
🙈